### DISCUSSION TITLE
Rest-API

### DISCUSSION THREAD

**Discussion by budda85** [OpenOlitor/OpenOlitor#125] [General]
Ich würde gerne nochmal bzgl. der API nachfragen, dass wir die besser dokumentieren.
Ich hatte ja [hier ](https://github.com/OpenOlitor/OpenOlitor/issues/79 )schon mal nachgefragt. Wenn man die API besser nutzen könnte, könnte man sich kleinere Scripts bauen um damit zu kommunizieren und ich müsste nicht direkt an der Datenbank rumfummeln.
Wenn mir einer erklärt wie man sich da anmelden kann, würde ich mich mal an eine Doku machen.
Wäre dann vielleicht auch super wenn man z. B. [Swagger UI](https://swagger.io/tools/swagger-ui/) integrieren kann.

**Comment by AlwinEgger** [comment]
Swagger wäre super. Mit der neuen Server-Architektur liesse sich dies auch machen.

Damit die Idee mit den Scripts gut funktioniert, müsste der Authentisierungsprozess aber auch noch ergänzt oder aktualisiert werden. Im Moment loggt sich ein User über die `/login` Route ein und erhält dann ein Cookie mit dem Session-Key mit der Response gesetzt. Dieses Cookie muss dann bei weiteren Calls auf die übrigen Routes gesetzt sein.

Die Generierung einer Form von technischen Access-Keys wäre eine Möglichkeit den Zugriff permanent möglich zu machen.

**Comment by budda85** [comment]
Das wäre super wenn das mit den Access-Keys implementiert werden würde.
Würde es auch einfacher machen wenn man 2FA/OTA nutzt.

**Comment by budda85** [comment]
Gibt es hier denn Pläne das umzusetzen?
